src/coreclr/jit/flowgraph.cpp

```diff
@@ -863,6 +863,9 @@ bool Compiler::fgAddrCouldBeNull(GenTree* addr)
         case GT_ARR_ADDR:
             return (addr->gtFlags & GTF_ARR_ADDR_NONNULL) == 0;
 
+        case GT_BOX:
+            return !addr->IsBoxedValue();
```

Comment:
remind me why GT_BOX *may* produce null possibly? I don't understand the need in `IsBoxedValue()`

Reply:
`IsBoxedValue` check the `GTF_BOX_VALUE` flag which is documented as guaranteeing the box is not null, when we discussed this on Discord we reached the conclusion that the flag might be an old leftover that's not needed anymore and could be removed but like I said there, I'd prefer such cleanup to be a separate PR.